### PR TITLE
Py3

### PR DESCRIPTION
**Pull request by matt-peters**
* Python 3 support
* prepare to distribute on pip
* old code removal

### CODE REVIEW

File: mozsci/ems.py [hunk 1]
```diff
@@ -56,9 +60,9 @@ def select_ensemble_bagged(self, y, ymodels, verbose=False):
         indices = np.arange(len(ymodels))
         max_keep = int(self.pbags * len(ymodels))
         nmodels = 0
-        for k in xrange(self.nbags):
+        for k in range(self.nbags):
             if verbose:
-                print("Bagging number %s" % str(k+1))
+                print(("Bagging number %s" % str(k+1)))
```
Comment (b4hand) [inline mozsci/ems.py]: You shouldn't need to double parenthesize this. That's only necessary if you don't explicitly import `print_function`.

File: mozsci/map_train.py [hunk 2]
```diff
@@ -89,7 +91,7 @@ def _run_kfold(self):
         if type(errors[0]['train']) == dict:
             for k in ['train', 'test']:
                 ret[k] = {}
-                for error_type in errors[0]['train'].keys():
+                for error_type in list(errors[0]['train'].keys()):
```
Comment (b4hand) [inline mozsci/map_train.py]: You shouldn't need to explicitly invoke `list` here. That will materialize the keys into a `list`, but it is already an iterable, so the `for` loop will just work.

File: mozsci/numpy_util.py [hunk 3]
```diff
@@ -14,7 +15,7 @@ def default(self, obj):
 def numpy_decoder(dct):
     """Decodes numpy arrays stored as values in a json dictionary
     Use like json.loads(j, object_hook=numpy_decoder)"""
-    for k in dct.keys():
+    for k in list(dct.keys()):
```
Comment (b4hand) [inline mozsci/numpy_util.py]: Same thing here, you can drop the `list` call.

File: test/test_map_train.py [hunk 4]
```diff
@@ -64,7 +65,7 @@ def test_run_train_models(self):
           [LogisticRegression, classification_error, None, (), {'lam':50}]]
 
         errors = run_train_models(2, model_library, X=self.X, y=self.y)
-        for k in errors.keys():
+        for k in list(errors.keys()):
```
Comment (b4hand) [inline test/test_map_train.py]: You can drop this call to `list`, but the ones in the asserts above are necessary.

**Comment by b4hand** [comment]
By the way, really happy to see this gaining Py3 support! :+1:

**Comment by matt-peters** [comment]
Thanks for the review.  FWIW, I used modernize to add most of the basic Python 3 stuff w/o much manual review except making sure the tests pass.  Removing the scipy.weave was most of the difficulty in making the upgrade.

**Comment by b4hand** [comment]
There were a few other places with the `print((` pattern. I would just look for those and replace all of them.

**Pull request merged**